universal-application-tool-0.0.1/app/filters/LoggingFilter.java

```diff
@@ -37,6 +45,31 @@ public EssentialAction apply(EssentialAction next) {
                     long time = clock.millis() - startTime;
                     log.info(
                         "{}\t{}\t{}ms\t{}", request.method(), request.uri(), time, result.status());
+                    StringBuilder requestCookies = new StringBuilder();
+                    Iterator<Http.Cookie> requestCookieIterator = request.cookies().iterator();
+                    while (requestCookieIterator.hasNext()) {
+                      Http.Cookie cookie = requestCookieIterator.next();
+                      requestCookies.append(
+                          String.format("key: %s, value: %s\n", cookie.name(), cookie.value()));
+                    }
+                    log.trace("request cookies: {}", requestCookies.toString());
+                    log.trace("response headers: {}", result.headers().toString());
+                    if (result.body() instanceof HttpEntity.Strict) {
```

Comment:
question: What is `HttpEntity.Strict` and why is it helpful to log?

Reply:
It means that the response is not streamed and therefore we have access to the whole thing - we can't log it otherwise since we don't have the whole thing.

Reply:
Neat! Learned something new today